Assets/Code/Behaviours/GenerateCity.cs

```diff
@@ -94,6 +117,40 @@ void Start()
         // https://docs.unity3d.com/ScriptReference/Transform-parent.html?_ga=2.51063529.1903334404.1586621367-266296751.1584777261 
     }
 
+    public MapCell GetCellFromPosition(Vector3 position)
+    {
+        var relativePosition = position - bottomLeftCornerPosition;
+        var row = Mathf.FloorToInt(relativePosition.x) / Mathf.RoundToInt(cellSize);
+        var col = Mathf.FloorToInt(relativePosition.z) / Mathf.RoundToInt(cellSize);
+        try
+        {
+            return _map.GetMapCell(row, col);
+        }
+        catch (Exception)

```

Comment:
Maybe we should add a check in the GetMapCell for the row and the col.
The exceptions are very slow mechanism, this should slow down the game quite a bit.

Reply:
you're right, moved to GetMapCell